### PR TITLE
Drop into collections fix

### PR DESCRIPTION
**Pull request by jamesgorrie**
A few small fixes that I noticed as single image drop wasn't working:
* Fix logic to work with single image drop data
* Make drop data with multiple images consistent with single image
* Make the saving state consistent for adding images to collections through drop or click

add-collection

### CODE REVIEW

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js [hunk 1]
```diff
@@ -138,48 +138,37 @@ grCollectionsPanel.directive('grDropIntoCollection',
...
-                            element.removeClass(classComplete);
-                        }, 500);
+                        scope.dropIntoCollectionSaving = false;
                     });
                 }
-                element.removeClass(classDrag);
+                element.removeClass(classOver);
             });
 
             element.on('dragover', () => {
```
Comment (theefer) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: Should it be drag enter or drag over?

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js [hunk 2]
```diff
@@ -138,48 +138,37 @@ grCollectionsPanel.directive('grDropIntoCollection',
...
+                if (gridImagesData !== '' || gridImageData !== '') {
                     const imagesData = gridImagesData !== '' ?
                         JSON.parse(gridImagesData) : [JSON.parse(gridImageData)];
 
-                    const imageIds = imagesData.map(imageJson => imageJson.id);
-
-                    // TODO: Find a better way of dealing with this state than classnames
-                    element.addClass(classSaving);
+                    const imageIds = imagesData.map(imageJson => imageJson.data.id);
+                    scope.dropIntoCollectionSaving = true;
```
Comment (theefer) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: Watch out with scope inheritance (just investigated a Composer bug due to use of scope in a non-scope-isolated directive with Rob yesterday). What scope is this on and does it risk to leak to siblings or children also observing it?

**Comment by theefer** [comment]
:+1: though a small warning re use of scope.

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: I suppose dragOver would make more sense.

File: kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js (hunk 2 above)
Comment (jamesgorrie) [inline kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js]: It's in the scope of the element that is the directive, it's not from the parent.

It's also only controlled from this directive, and read from within the element that has this as an attribute, thus the template / element for this directive.

I can't see any leak there, unless I am being :see_no_evil:

**Pull request merged**